src/Filter.test.js

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import Filter from './Filter';
+import { shallow } from 'enzyme';
+
+  const mockColoradoClimbingAreas = {
+    "Canon City": {
+        "milesFromDenver": 119,
+        "id": 34,
+        "typesOfClimbing": ["trad", "sport", "toprope", "boulder"],
+        "classicRoutes": [
+            {
+                "name": "Enterprise",
+                "id": 35,
+                "grade": "5.09+",
+                "type": ["sport"],
+                "link": "https://www.mountainproject.com/route/105824687/enterprise"
+            },
+            {
+                "name": "Trick-or-Treat",
+                "id": 42,
+                "grade": "5.11a/b",
+                "type": ["toprope"],
+                "link": "https://www.mountainproject.com/route/105759012/trick-or-treat"
+            }
+        ]
+    },
+    "Boulder Canyon": {
+        "milesFromDenver": 31.5,
+            "id": 12,
+            "typesOfClimbing": ["trad", "sport", "toprope", "boulder", "aid"],
+            "classicRoutes": [
+                {
+                    "name": "North Face Center",
+                    "id": 13,
+                    "grade": "5.07+",
+                    "type": ["trad"],
+                    "link": "https://www.mountainproject.com/route/105748093/north-face-center"
+                },
+                {
+                    "name": "Empor",
+                    "id": 14,
+                    "grade": "5.07+",
+                    "type": ["trad"],
+                    "link": "https://www.mountainproject.com/route/105748142/empor"
+                }
+            ]
+        }
+    }
+
+
+  const mockFunction = jest.fn();
+
+
+
+
+
+
+describe('Filter', () => {
+
+  let wrapper;
+
+
+  beforeEach(() => {
+    wrapper = shallow(
+      <Filter climb={mockColoradoClimbingAreas}
+                    updateFromFilter={mockFunction}/>
+      )
+  })
+
+  it('should test the states', () => {
+    expect(wrapper.state()).toEqual({setMax: 50, types: [], areas: [], areasPick: [], typesPick: []
+    })
+  })
+
+  it('should change type', () => {
+    expect(wrapper.state('setMax')).toEqual(50);
+    expect(wrapper.state('types')).toEqual([]);
+    wrapper.find('.range').simulate('change', {target: {value: 120}});
+    expect(wrapper.state('setMax')).toEqual(120);
+    expect(wrapper.state('types')).toEqual(["trad", "sport", "toprope", "boulder", "aid"]);
+  })
+
+  it('should change Area', () => {
+    expect(wrapper.state('typesPick')).toEqual([]);
+    expect(wrapper.state('areas')).toEqual([]);
+    wrapper.find('.range').simulate('change', {target: {value: 120}});
+    wrapper.find('.types-checkbox').at(0).simulate('change', {target: {checked: true}})
```

Comment:
Ok, I see that we are trying to test one checkbox here using 'at(0)'. So would we just be testing the impact of checking one box (at index 0) even though the user can select multiple checkboxes?

Reply:
not sure. I need help fixing this in the AM. trying to simulate the change seems to break the test every time.